lib/iteration.rb

```diff
@@ -0,0 +1,109 @@
+def benchmark(num = 1)
+  start = Time.now
+  num.times { yield }
+  elapsed_time = (Time.now - start) / num
+end
+
+def race(procs, num = 1)
+  arr = []
+  procs.each do |proc|
+    runtime = benchmark(num)
+    arr << runtime
+  end
+  index = arr.index(arr.max)
+  procs[index]
+end
+
+def eval_block(*args, &prc)
+  if prc.nil?
+    "NO BLOCK GIVEN"
+  else
+    p yield *args
+  end
+end
+
+def factors(num)
+  (1..num).select { |n| num % n == 0 }
+end
+
+class Array
+  def bubble_sort!(&prc)
```

Comment:
Good

On Tue, May 31, 2016 at 12:31 PM, Jeff Liu <notifications@github.com> wrote:

> In lib/iteration.rb
> <https://github.com/emperorliu/test-first-ruby/pull/4#discussion_r65247585>
> :
>
> > +end
> > +
> > +def eval_block(*args, &prc)
> > +  if prc.nil?
> > +    "NO BLOCK GIVEN"
> > +  else
> > +    p yield *args
> > +  end
> > +end
> > +
> > +def factors(num)
> > +  (1..num).select { |n| num % n == 0 }
> > +end
> > +
> > +class Array
> > +  def bubble_sort!(&prc)
>
> Yes. You sort by comparing the current element to the one after, and keep
> going until the end of the array. Then you repeat. The largest number would
> naturally get sorted to the end the first round.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/emperorliu/test-first-ruby/pull/4/files/dd2b0b193a8a825ec50c28358bc98518685c32d5#r65247585>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABtEnCtvOh2CaMspxEQjzeRNrddgyyxcks5qHIyFgaJpZM4IqHIJ>
> .
>